pkg/prometheus/agent/operator.go

```diff
@@ -808,6 +808,21 @@ func (c *Operator) UpdateStatus(ctx context.Context, key string) error {
 	}
 	p.Status = *pStatus
 
+	selectorLabels := map[string]string{
+		"app.kubernetes.io/name":       "prometheus-agent",
+		"app.kubernetes.io/managed-by": "prometheus-operator",
+		"app.kubernetes.io/instance":   p.Name,
+	}
```

Comment:
maybe we should have a shared function with `makeStatefulSetSpec()` to return the selector labels?

Reply:
> maybe we should have a shared function with `makeStatefulSetSpec()` to return the selector labels?

I'm not sure I understand what you mean by "shared function with makeStatefulSetSpec"...

could you elaborate?

Reply:
I'm referring to this part where the controller defines the label selector for the statefulset's pods:
https://github.com/prometheus-operator/prometheus-operator/blob/a8eb064ec76327afca43638085c82a20f6850689/pkg/prometheus/server/statefulset.go#L347-L354

Reply:
addressed in [0b2ecbf](https://github.com/prometheus-operator/prometheus-operator/pull/5962/commits/0b2ecbf142aed30dc423236da5576467b8bc0fbc)

thanks for clarifying